pkg/collector/skipper_collector.go

```diff
@@ -1,20 +1,24 @@
 package collector
 
 import (
+	"encoding/json"
 	"fmt"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"math"
 	"strings"
 	"time"
 
 	autoscalingv2beta1 "k8s.io/api/autoscaling/v2beta1"
-	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/metrics/pkg/apis/custom_metrics"
 )
 
 const (
-	rpsQuery      = `scalar(sum(rate(skipper_serve_host_duration_seconds_count{host="%s"}[1m])))`
-	rpsMetricName = "requests-per-second"
+	rpsQuery                  = `scalar(sum(rate(skipper_serve_host_duration_seconds_count{host="%s"}[1m])))`
+	rpsMetricName             = "requests-per-second"
+	stacksetWeightsAnnotation = "zalando.org/stack-traffic-weights"
+	backendWeightsAnnotation  = "zalando.org/backend-weights"
```

Comment:
I'm not sure it makes sense to use a max over these two values as the weight because `zalando.org/stack-traffic-weights` is only the desired value while `zalando.org/backend-weights` is the actual current value and therefore more precisely represents the weight that the particular stack will be getting. Using only `zalando.org/backend-weights` would also limit this project to only depend on skipper and not have any relation to stackset-controller (only the other way around which is fine).

WDYT?

Reply:
It does rely on skipper doesn't it? Because it's unlikely that anything else will expose those metrics with that name or is it a standard. Also I take the max over the two because say while scaling up the stackset-controller annotation is changed, but the skipper pods are still using the backend annotations. Scaling down the old stackset at that point may result in requests being dropped. That said it may make sense to pass the annotations as flags instead of hardcoding it as constants.

Reply:
Pass a flag makes most sense and set a default to `zalando.org/backend-weights`. Like this it works out of the box with skipper, but also traefik users can use it with their custom annotation.

Reply:
If we use only `backend-weights` won't the scale-up only happen once the traffic has been switched? I think it may cause issues because existing pods won't be able to handle the new load.

Reply:
I just don't think we can rely purely on `max( stack-traffic-weights, backend-weights)` for pre-scaling because it will anyway take some time for the HPA to react, and the stack might get traffic before the HPA has scaled. This is currently handled by pre-scaling in stackset-controller, which i think is fine.

That being said `max( stack-traffic-weights, backend-weights)` should not hurt as it would in worst case scale a bit too high. So I'm fine with `max( stack-traffic-weights, backend-weights)` based on our discussion here! :+1: